idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtOverrideMembersHandler.kt

```diff
@@ -0,0 +1,107 @@
+            )
+        }
+    }
+
+    @OptIn(ExperimentalStdlibApi::class)
+    private fun KtAnalysisSession.getOverridableMembers(classOrObjectSymbol: KtClassOrObjectSymbol): List<OverrideMember> {
+        return buildList {
+            classOrObjectSymbol.getMemberScope().getCallableSymbols().forEach { symbol ->
+                if (!symbol.isVisibleInClass(classOrObjectSymbol)) return@forEach
+                val implementationStatus = symbol.getImplementationStatus(classOrObjectSymbol) ?: return@forEach
+                if (!implementationStatus.isOverridable) return@forEach
+
+                val intersectionSymbols = symbol.intersectionSymbols
+                val symbolsToProcess = if (intersectionSymbols?.isNotEmpty() == true) {
+                    val nonAbstractMembers = intersectionSymbols.filter { (it as? KtSymbolWithModality)?.modality != Modality.ABSTRACT }
+                    // If there are non-abstract members, we only want to show override for these non-abstract members. Otherwise, show any
+                    // abstract member to override.
+                    nonAbstractMembers.ifEmpty {
+                        listOf(intersectionSymbols.first())
+                    }
+                } else {
+                    listOf(symbol)
+                }
+
+                for (symbolToProcess in symbolsToProcess) {
+                    val originalOverriddenSymbol = symbolToProcess.originalOverriddenSymbol
+                    val containingSymbol = originalOverriddenSymbol?.originalContainingClassForOverride
+
+                    val bodyType = when {
+                        classOrObjectSymbol.classKind == KtClassKind.INTERFACE && containingSymbol?.classIdIfNonLocal == StandardClassIds.Any -> {
+                            if (classOrObjectSymbol.superTypes.singleOrNull()?.type?.isAny == true) {
+                                // If an interface does not extends any other interfaces, FE1.0 simply skips members of `Any`. So we mimic
+                                // the same behavior. See idea/testData/codeInsight/overrideImplement/noAnyMembersInInterface.kt
+                                continue
+                            } else {
+                                BodyType.NO_BODY
+                            }
+                        }
+                        (originalOverriddenSymbol as? KtSymbolWithModality)?.modality == Modality.ABSTRACT ->
+                            BodyType.FROM_TEMPLATE
+                        symbolsToProcess.size > 1 ->
+                            BodyType.QUALIFIED_SUPER
+                        else ->
+                            BodyType.SUPER
+                    }
+                    // Ideally, we should simply create `KtClassMember` here and remove the intermediate `OverrideMember` data class. But
+                    // that doesn't work because this callback function is holding a read lock and `symbol.render(renderOption)` requires
```

Comment:
Don't understand this comment... simply creating `KtClassMember` doesn't actually do any rendering until `generateMember` is called

Reply:
It does because `KtClassMember` needs the rendered text for the popup dialog where user can pick elements to implement/override.